packages/api/src/dataProviders/db/programme/programme.port.ts

```diff
@@ -0,0 +1,22 @@
+import MongoRepository from "../../../shared/MongoRepository";
+import ProgrammeDbo from "./ProgrammeDbo";
+
+export class ProgrammePort extends MongoRepository<ProgrammeDbo> {
+    collectionName = "programme";
+
+    public createIndexes(): void {
+        this.collection.createIndex({ code: 1 }, { unique: true });
+    }
+
+    public findByCode(code: string): Promise<ProgrammeDbo> {
+        return this.collection.find({ code }).toArray()[0];
+    }
+
+    public async replace(programmes: ProgrammeDbo[]) {
```

Comment:
```suggestion
    public async replaceAll(programmes: ProgrammeDbo[]) {
```
```suggestion
    public async resync(programmes: ProgrammeDbo[]) {
```
c'était peut être déjà là mais `replace` ça me donne l'impression que tu en mets à jour un en particulier